SingularityService/src/main/java/com/hubspot/singularity/mesos/SingularityMesosOfferScheduler.java

```diff
@@ -258,22 +261,61 @@ private double score(SingularityOfferHolder offerHolder, SingularitySchedulerSta
 
     SingularitySlaveUsageWithId slaveUsage = maybeSlaveUsage.get();
-    Map<RequestType, Map<ResourceUsageType, Number>> usagesPerRequestType = slaveUsage.getUsagePerRequestType();
-    Map<ResourceUsageType, Number> usagePerResource = usagesPerRequestType.get(taskRequest.getRequest().getRequestType());
+    Map<ResourceUsageType, Number> longRunningTasksUsage = slaveUsage.getLongRunningTasksUsage();
+
+    double longRunningCpusUsedScore = longRunningTasksUsage.get(ResourceUsageType.CPU_USED).doubleValue() / slaveUsage.getCpuTotal().get();
+    double longRunningMemUsedScore = ((double) longRunningTasksUsage.get(ResourceUsageType.MEMORY_BYTES_USED).longValue() / slaveUsage.getMemoryBytesTotal().get());
+
+    double cpusTotalScore = (MesosUtils.getNumCpus(offer) / slaveUsage.getCpuTotal().get());
+    double memTotalScore = (MesosUtils.getMemory(offer) / slaveUsage.getMemoryMbTotal().get());
+
+    return isLongRunning(taskRequest) ? scoreLongRunningTask(longRunningMemUsedScore, memTotalScore, longRunningCpusUsedScore, cpusTotalScore)
+        : scoreNonLongRunningTask(taskRequest, longRunningMemUsedScore, memTotalScore, longRunningCpusUsedScore, cpusTotalScore);
+  }
+
+  private boolean isLongRunning(SingularityTaskRequest taskRequest) {
+    return taskRequest.getRequest().getRequestType().isLongRunning();
+  }
+
+  private double scoreLongRunningTask(double longRunningMemUsedScore, double memTotalScore, double longRunningCpusUsedScore, double cpusTotalScore) {
+    // usage improves score
+    return calculateScore(1 - longRunningMemUsedScore, memTotalScore, 1 - longRunningCpusUsedScore, cpusTotalScore, 0.50, 0.50);
+  }
+
+  private double scoreNonLongRunningTask(SingularityTaskRequest taskRequest, double longRunningMemUsedScore, double memTotalScore, double longRunningCpusUsedScore, double cpusTotalScore) {
+    Optional<SingularityDeployStatistics> statistics = deployManager.getDeployStatistics(taskRequest.getRequest().getId(), taskRequest.getDeploy().getId());
+
+    double freeResourceWeight = 0.75;
+    double usedResourceWeight = 0.25;
+
+    if (statistics.isPresent() && statistics.get().getAverageRuntimeMillis().isPresent()) {
+      final double maxNonLongRunningUsedResourceWeight = configuration.getMaxNonLongRunningUsedResourceWeight();
+      usedResourceWeight = Math.min((double) TimeUnit.MILLISECONDS.toSeconds(statistics.get().getAverageRuntimeMillis().get()) / configuration.getConsiderNonLongRunningTaskLongRunningAfterRunningForSeconds(), 1) * maxNonLongRunningUsedResourceWeight;
+
+      if (usedResourceWeight == maxNonLongRunningUsedResourceWeight) {
+        return scoreLongRunningTask(longRunningMemUsedScore, memTotalScore, longRunningCpusUsedScore, cpusTotalScore);
+      }
+      freeResourceWeight = 1 - usedResourceWeight;
+    }
+
+    // usage reduces score
+    return calculateScore(longRunningMemUsedScore, memTotalScore, longRunningCpusUsedScore, cpusTotalScore, freeResourceWeight, usedResourceWeight * -1);
+  }
```

Comment:
These changes will be easiest to discuss in person.

Essentially, long running tasks consider an offer 50:50 for usage and free space. non long running offers vary depending on how long they've been running with a default at the middle (25:75). non long running offers favor free space and will score lower if they run for a large amount of time **and** there are long running tasks on the  offered machine